cmrf_views/src/Plugin/views/query/API4.php

```diff
@@ -0,0 +1,525 @@
+      $alias = $this->view->storage->get('base_field');
+    }
+
+    if (!$alias && $table) {
+      $alias = $table . '_' . $field;
+    }
+
+    // Make sure an alias is assigned
+    $alias = $alias ? $alias : $field;
+
+    // Create a field info array.
+    $field_info = [
+        'field' => $field,
+        'table' => $table,
+        'alias' => $alias,
+      ] + $params;
+
+    // Test to see if the field is actually the same or not. Due to
+    // differing parameters changing the aggregation function, we need
+    // to do some automatic alias collision detection:
+    $base = $alias;
+    $counter = 0;
+    while (!empty($this->fields[$alias]) && $this->fields[$alias] != $field_info) {
+      $field_info['alias'] = $alias = $base . '_' . ++$counter;
+    }
+
+    if (empty($this->fields[$alias])) {
+      $this->fields[$alias] = $field_info;
+    }
+
+    // Keep track of all aliases used.
+    $this->fieldAliases[$table][$field] = $alias;
+
+    return $alias;
+  }
+
+  /**
+   * Returns the alias for the given field added to $table.
+   *
+   * @access protected
+   *
+   * @see \Drupal\cmrf_views\Plugin\views\query\API::addField()
+   */
+  protected function getFieldAlias($table_alias, $field) {
+    $field = str_replace('.', '__', $field);
+    return isset($this->fieldAliases[$table_alias][$field]) ? $this->fieldAliases[$table_alias][$field] : FALSE;
+  }
+
+  /**
+   * Executes the query and fills the associated view object with according
+   * values.
+   *
+   * Values to set: $view->result, $view->total_rows, $view->execute_time,
+   * $view->current_page.
+   */
+  public function execute(ViewExecutable $view) {
+    $table_data = $this->viewsData->get($view->storage->get('base_table'));
+    if (!empty($table_data)) {
+      $api_entity       = $table_data['table']['base']['entity'];
+      $api_action       = $table_data['table']['base']['action'];
+      $api_version      = $table_data['table']['base']['api_version'];
+      $connector        = $table_data['table']['base']['connector'];
+      $dataset_params   = $table_data['table']['base']['params'] ?? [];
+      if (!is_array($dataset_params)) {
+        $dataset_params = [];
+      }
+
+      $parameters = [];
+      $start      = microtime(TRUE);
+
+      // Set the return fields
+      $parameters['select'] = [];
+      foreach ($view->field as $field) {
+        if (!empty($table_data[$field->field]['cmrf_original_definition']['name'])) {
+          $original_field_name = $table_data[$field->field]['cmrf_original_definition']['name'];
+          if (!in_array($original_field_name, $parameters['return'])) {
+            $parameters['select'][] = $original_field_name;
+          }
+        }
+      }
+
+      // Set the query parameters.
+      if (!empty($this->where)) {
+        foreach ($this->where as $where_group) {
+          if (empty($where_group['conditions'])) {
```

Comment:
Maybe this `if` could be removed if the `foreach` below is changed to `foreach ($where_group['conditions'] ?? [] as $condition)`